Makefile

```diff
@@ -65,9 +65,10 @@ check-dependencies:
 lint-all: lint-js lint-python lint-html lint-docs lint-docker lint-actions lint-shell
 
 lint: lint-js lint-python lint-html lint-links
+# prettier: delete the cache when updating plugins, eslint: no need to delete the cache
```

Comment:
Is this an instruction for PL developers? A description of what the tools will do? I don't understand why this is here.

Reply:
Perhaps I can put this developer note in the prettierrc?

Reply:
I take that to mean that these are instructions? Your description of how Prettier works worries me, and that feels huge footgun for several reasons:

- No one is going to think to look at either the Makefile or `.prettierrc` after updating a Prettier plugin (or a dependency of such, like `sql-formatter`.
- In CI, the use of `restore-keys` without a content hash means that in most cases, we'll unconditionally restore a cache that might have outdated information.

I'm pretty hesitant to add this.

Reply:
I opened an upstream issue to talk about this here: https://github.com/prettier/prettier/issues/17260. While I agree it's not ideal -- and I can fixup CI to avoid this by only using a restore key based off yarn.lock on cache miss, the local speedup for me is really nice. I'll mark this as blocked until something happens on that issue.

Reply:
Caching based on version would still be unlikely to help us, at least in the case of the SQL formatter, as the version is only ever changed by `changesets` well after a change has been merged. Of course, we could make the version be a hash of the implementation and lockfile, I guess? That seems like a lot of complexity though.

Meta question: do you actually find yourself regularly running `make format-js`? Our VS Code config will auto-format files on save, so I basically never do anything to explicitly format the entire codebase. If you do, I'm curious to hear why, maybe there's a different rough edge we should work on.

Reply:
That's a good point -- I'll double check my json/yaml formatter settings, that's where I run into issues.

Reply:
Here's the relevant piece of our VS Code settings that should be enabling Prettier formatting for those files:

https://github.com/PrairieLearn/PrairieLearn/blob/a553a3cb4298da44fab4cf2e718deef5dea39480/.vscode/settings.json#L28-L31

Reply:
The restore key now follows your suggestion to correctly invalidate the cache, and I removed the Makefile note.